### ISSUE TITLE
Node 12 support

### ISSUE THREAD

**Issue by bmesuere** [tree-sitter/tree-sitter-javascript#124]
I can't seem to install this using Node 12, both on my macbook as on GitHub Actions. I get the following errors from node-gyp:

```
2019-10-27T16:03:53.1534487Z Command: node-gyp rebuild
2019-10-27T16:03:53.1534807Z Arguments: 
2019-10-27T16:03:53.1535595Z Directory: /home/runner/work/dolos/dolos/node_modules/tree-sitter-javascript
2019-10-27T16:03:53.1536024Z Output:
2019-10-27T16:03:53.1536485Z gyp info it worked if it ends with ok
2019-10-27T16:03:53.1538251Z gyp info using node-gyp@5.0.5
2019-10-27T16:03:53.1538717Z gyp info using node@12.13.0 | linux | x64
2019-10-27T16:03:53.1539138Z gyp info find Python using Python version 2.7.15 found at "/usr/bin/python"
2019-10-27T16:03:53.1539984Z gyp http GET https://nodejs.org/download/release/v12.13.0/node-v12.13.0-headers.tar.gz
2019-10-27T16:03:53.1541456Z gyp http 200 https://nodejs.org/download/release/v12.13.0/node-v12.13.0-headers.tar.gz
2019-10-27T16:03:53.1542024Z gyp http GET https://nodejs.org/download/release/v12.13.0/SHASUMS256.txt
2019-10-27T16:03:53.1542419Z gyp http 200 https://nodejs.org/download/release/v12.13.0/SHASUMS256.txt
2019-10-27T16:03:53.1542726Z gyp info spawn /usr/bin/python
2019-10-27T16:03:53.1542982Z gyp info spawn args [
2019-10-27T16:03:53.1543882Z gyp info spawn args   '/opt/hostedtoolcache/node/12.13.0/x64/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
2019-10-27T16:03:53.1544640Z gyp info spawn args   'binding.gyp',
2019-10-27T16:03:53.1545181Z gyp info spawn args   '-f',
2019-10-27T16:03:53.1545721Z gyp info spawn args   'make',
2019-10-27T16:03:53.1546230Z gyp info spawn args   '-I',
2019-10-27T16:03:53.1547649Z gyp info spawn args   '/home/runner/work/dolos/dolos/node_modules/tree-sitter-javascript/build/config.gypi',
2019-10-27T16:03:53.1548287Z gyp info spawn args   '-I',
2019-10-27T16:03:53.1549158Z gyp info spawn args   '/opt/hostedtoolcache/node/12.13.0/x64/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
2019-10-27T16:03:53.1550312Z gyp info spawn args   '-I',
2019-10-27T16:03:53.1551316Z gyp info spawn args   '/home/runner/.cache/node-gyp/12.13.0/include/node/common.gypi',
2019-10-27T16:03:53.1552312Z gyp info spawn args   '-Dlibrary=shared_library',
2019-10-27T16:03:53.1553222Z gyp info spawn args   '-Dvisibility=default',
2019-10-27T16:03:53.1554397Z gyp info spawn args   '-Dnode_root_dir=/home/runner/.cache/node-gyp/12.13.0',
2019-10-27T16:03:53.1555193Z gyp info spawn args   '-Dnode_gyp_dir=/opt/hostedtoolcache/node/12.13.0/x64/lib/node_modules/npm/node_modules/node-gyp',
2019-10-27T16:03:53.1556093Z gyp info spawn args   '-Dnode_lib_file=/home/runner/.cache/node-gyp/12.13.0/<(target_arch)/node.lib',
2019-10-27T16:03:53.1558144Z gyp info spawn args   '-Dmodule_root_dir=/home/runner/work/dolos/dolos/node_modules/tree-sitter-javascript',
2019-10-27T16:03:53.1558818Z gyp info spawn args   '-Dnode_engine=v8',
2019-10-27T16:03:53.1559457Z gyp info spawn args   '--depth=.',
2019-10-27T16:03:53.1560075Z gyp info spawn args   '--no-parallel',
2019-10-27T16:03:53.1561403Z gyp info spawn args   '--generator-output',
2019-10-27T16:03:53.1562733Z gyp info spawn args   'build',
2019-10-27T16:03:53.1566066Z gyp info spawn args   '-Goutput_dir=.'
2019-10-27T16:03:53.1566634Z gyp info spawn args ]
2019-10-27T16:03:53.1568100Z gyp info spawn make
2019-10-27T16:03:53.1568819Z gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
2019-10-27T16:03:53.1569662Z make: Entering directory '/home/runner/work/dolos/dolos/node_modules/tree-sitter-javascript/build'
2019-10-27T16:03:53.1570128Z   CC(target) Release/obj.target/tree_sitter_javascript_binding/src/parser.o
2019-10-27T16:03:53.1570524Z   CC(target) Release/obj.target/tree_sitter_javascript_binding/src/scanner.o
2019-10-27T16:03:53.1570898Z   CXX(target) Release/obj.target/tree_sitter_javascript_binding/src/binding.o
2019-10-27T16:03:53.1571899Z ../src/binding.cc:13:11: error: variable or field â€˜Initâ€™ declared void
2019-10-27T16:03:53.1572454Z  void Init(Handle<Object> exports, Handle<Object> module) {
2019-10-27T16:03:53.1572956Z            ^~~~~~
2019-10-27T16:03:53.1575894Z ../src/binding.cc:13:11: error: â€˜Handleâ€™ was not declared in this scope
2019-10-27T16:03:53.1577616Z ../src/binding.cc:13:11: note: suggested alternative: â€˜rand_râ€™
2019-10-27T16:03:53.1578055Z  void Init(Handle<Object> exports, Handle<Object> module) {
2019-10-27T16:03:53.1578313Z            ^~~~~~
2019-10-27T16:03:53.1578591Z            rand_r
2019-10-27T16:03:53.1579324Z ../src/binding.cc:13:24: error: expected primary-expression before â€˜>â€™ token
2019-10-27T16:03:53.1579780Z  void Init(Handle<Object> exports, Handle<Object> module) {
2019-10-27T16:03:53.1580497Z                         ^
2019-10-27T16:03:53.1583960Z ../src/binding.cc:13:26: error: â€˜exportsâ€™ was not declared in this scope
2019-10-27T16:03:53.1584231Z  void Init(Handle<Object> exports, Handle<Object> module) {
2019-10-27T16:03:53.1584370Z                           ^~~~~~~
2019-10-27T16:03:53.1584855Z ../src/binding.cc:13:35: error: â€˜Handleâ€™ was not declared in this scope
2019-10-27T16:03:53.1585058Z  void Init(Handle<Object> exports, Handle<Object> module) {
2019-10-27T16:03:53.1585262Z                                    ^~~~~~
2019-10-27T16:03:53.1585711Z ../src/binding.cc:13:35: note: suggested alternative: â€˜rand_râ€™
2019-10-27T16:03:53.1585872Z  void Init(Handle<Object> exports, Handle<Object> module) {
2019-10-27T16:03:53.1586214Z                                    ^~~~~~
2019-10-27T16:03:53.1586392Z                                    rand_r
2019-10-27T16:03:53.1587552Z ../src/binding.cc:13:48: error: expected primary-expression before â€˜>â€™ token
2019-10-27T16:03:53.1587930Z  void Init(Handle<Object> exports, Handle<Object> module) {
2019-10-27T16:03:53.1588158Z                                                 ^
2019-10-27T16:03:53.1588713Z ../src/binding.cc:13:50: error: â€˜moduleâ€™ was not declared in this scope
2019-10-27T16:03:53.1588937Z  void Init(Handle<Object> exports, Handle<Object> module) {
2019-10-27T16:03:53.1589172Z                                                   ^~~~~~
2019-10-27T16:03:53.1589628Z ../src/binding.cc:13:50: note: suggested alternative: â€˜mode_tâ€™
2019-10-27T16:03:53.1589847Z  void Init(Handle<Object> exports, Handle<Object> module) {
2019-10-27T16:03:53.1590081Z                                                   ^~~~~~
2019-10-27T16:03:53.1590319Z                                                   mode_t
2019-10-27T16:03:53.1590548Z In file included from ../src/binding.cc:2:0:
2019-10-27T16:03:53.1591171Z ../src/binding.cc:26:45: error: â€˜Initâ€™ was not declared in this scope
2019-10-27T16:03:53.1591354Z  NODE_MODULE(tree_sitter_javascript_binding, Init)
2019-10-27T16:03:53.1591708Z                                              ^
2019-10-27T16:03:53.1592320Z /home/runner/.cache/node-gyp/12.13.0/include/node/node.h:560:36: note: in definition of macro â€˜NODE_MODULE_Xâ€™
2019-10-27T16:03:53.1592584Z        (node::addon_register_func) (regfunc),                          \
2019-10-27T16:03:53.1592731Z                                     ^~~~~~~
2019-10-27T16:03:53.1593156Z ../src/binding.cc:26:1: note: in expansion of macro â€˜NODE_MODULEâ€™
2019-10-27T16:03:53.1593440Z  NODE_MODULE(tree_sitter_javascript_binding, Init)
2019-10-27T16:03:53.1593590Z  ^~~~~~~~~~~
2019-10-27T16:03:53.1593800Z ../src/binding.cc:26:45: note: suggested alternative:
2019-10-27T16:03:53.1593933Z  NODE_MODULE(tree_sitter_javascript_binding, Init)
2019-10-27T16:03:53.1594116Z                                              ^
2019-10-27T16:03:53.1594689Z /home/runner/.cache/node-gyp/12.13.0/include/node/node.h:560:36: note: in definition of macro â€˜NODE_MODULE_Xâ€™
2019-10-27T16:03:53.1594934Z        (node::addon_register_func) (regfunc),                          \
2019-10-27T16:03:53.1595081Z                                     ^~~~~~~
2019-10-27T16:03:53.1595508Z ../src/binding.cc:26:1: note: in expansion of macro â€˜NODE_MODULEâ€™
2019-10-27T16:03:53.1595687Z  NODE_MODULE(tree_sitter_javascript_binding, Init)
2019-10-27T16:03:53.1595816Z  ^~~~~~~~~~~
2019-10-27T16:03:53.1596601Z /home/runner/.cache/node-gyp/12.13.0/include/node/node.h:217:18: note:   â€˜node::Initâ€™
2019-10-27T16:03:53.1596925Z  NODE_EXTERN void Init(int* argc,
2019-10-27T16:03:53.1597462Z                   ^~~~
2019-10-27T16:03:53.1597696Z In file included from ../src/binding.cc:3:0:
2019-10-27T16:03:53.1598703Z ../src/binding.cc:11:12: warning: â€˜Nan::NAN_METHOD_RETURN_TYPE {anonymous}::New(Nan::NAN_METHOD_ARGS_TYPE)â€™ defined but not used [-Wunused-function]
2019-10-27T16:03:53.1598917Z  NAN_METHOD(New) {}
2019-10-27T16:03:53.1599030Z             ^
2019-10-27T16:03:53.1599523Z ../../nan/nan.h:1566:33: note: in definition of macro â€˜NAN_METHODâ€™
2019-10-27T16:03:53.1599790Z      Nan::NAN_METHOD_RETURN_TYPE name(Nan::NAN_METHOD_ARGS_TYPE info)
2019-10-27T16:03:53.1599994Z                                  ^~~~
2019-10-27T16:03:53.1600852Z tree_sitter_javascript_binding.target.mk:115: recipe for target 'Release/obj.target/tree_sitter_javascript_binding/src/binding.o' failed
2019-10-27T16:03:53.1601280Z make: *** [Release/obj.target/tree_sitter_javascript_binding/src/binding.o] Error 1
2019-10-27T16:03:53.1601807Z make: Leaving directory '/home/runner/work/dolos/dolos/node_modules/tree-sitter-javascript/build'
2019-10-27T16:03:53.1601993Z gyp ERR! build error 
2019-10-27T16:03:53.1602201Z gyp ERR! stack Error: `make` failed with exit code: 2
2019-10-27T16:03:53.1602837Z gyp ERR! stack     at ChildProcess.onExit (/opt/hostedtoolcache/node/12.13.0/x64/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
2019-10-27T16:03:53.1603096Z gyp ERR! stack     at ChildProcess.emit (events.js:210:5)
2019-10-27T16:03:53.1603383Z gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
2019-10-27T16:03:53.1603730Z gyp ERR! System Linux 5.0.0-1020-azure
2019-10-27T16:03:53.1604451Z gyp ERR! command "/opt/hostedtoolcache/node/12.13.0/x64/bin/node" "/opt/hostedtoolcache/node/12.13.0/x64/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
2019-10-27T16:03:53.1604952Z gyp ERR! cwd /home/runner/work/dolos/dolos/node_modules/tree-sitter-javascript
2019-10-27T16:03:53.1605288Z gyp ERR! node -v v12.13.0
2019-10-27T16:03:53.1605622Z gyp ERR! node-gyp -v v5.0.5
2019-10-27T16:03:53.1605772Z gyp ERR! not ok
```

**Comment by maxbrunsfeld** [comment]
Sorry for the delay. Fixed in 0.15.2.

**Comment by 3nuc** [comment]
I wanted to write a report that node-gyp rebuild fails with Node 14 but then I switched to 12 and it built fine. Then I switched back to Node **14.16.0** (with nvm install --lts) and it ... suddenly built there too.

If anyone stumbles here the solution may be "update your node to latest minor version"
